Arelle/Arelle
# Conformance suite assets configuration

_Review by commit_

___

#### Reason for change
The catalyst for this is the soon-to-be-implemented #797. These suites will be multi-GB files. This is problematic since each of our conformance suite CI jobs sync the full private conformance suite directory from S3.

One obvious solution (that hasn't been critical thus far) is to only download the conformance suite that each job is actually running. I wanted to pull this step from where it currently resides in the GHA workflow into the actual conformance suite runner. While considering the addition of *another* step in the conformance suite runner that downloads external assets, I decided that an overhaul was necessary for how we set up assets in our conformance suite configs.

The solution also happens to solve several other issues, including no longer needing to run public conformance suites in the `integration-tests` environment and thus requiring manual approval.

#### Description of change
* Implement `ConformanceSuiteAssetConfig` 
* Configure existing conformance suite assets with the new config approach
* Update CI to no longer use perform S3 sync and only conditionally require/setup AWS credentials

___

_Follow up: move publicly available conformance suites into the `arelle-public` bucket and configure the associated assets to be non-private. Until then, the only conformance suites affected by this are the suites included in the Arelle codebase._


#### Steps to Test
* CI 

Changes to CI won't appear on this PR (running on fork), but you can check these runs on my fork that demonstrate [non-private conformance suites running without approval](https://github.com/aaroncameron-wk/arelle-public/actions/runs/8561661056) and [successful runs with approval using the new download approach](https://github.com/aaroncameron-wk/arelle-public/actions/runs/8561661063).

**review**:
@Arelle/arelle
